spec/controllers/api/v1/users_controller_spec.rb

```diff
@@ -258,6 +239,40 @@
               expect(result_id).to eq(hard_user.id.to_s)
             end
           end
+
+          context 'with invalid search characters' do
+            let(:index_options){ { search: '@some&@!_(   )user' } }
+
+            it 'should strip the invalid characters' do
+              expect_any_instance_of(User::ActiveRecord_Relation)
+                .to receive(:full_search_login).with('some_user').and_call_original
+              get :index, index_options
+            end
+          end
+
+          context 'with a short search string' do
+            let(:index_options){ { search: 'me' } }
+
+            it 'should abort the query', :aggregate_failures do
+              expect_any_instance_of(User::ActiveRecord_Relation)
+                .to_not receive(:full_search_login)
+              expect(User).to receive(:none).and_call_original
+              get :index, index_options
+            end
+
+            # Why doesn't this pass?
+            #   login_search = query.where("lower(login) = ?", search_names.downcase)
+            # That expression doesn't modify the query criteria???
```

Comment:
I wonder if it's a result of UserSearchString inheriting from String. That usually doesn't work out too well.